### PR TITLE
Implemented Login and Signup functionality

### PR DESCRIPTION
**Pull request by ShubhamTiwari55**
**Summary**
This PR includes the following features and enhancements:

**User Interface Implementation:**

Designed and implemented the UI for the signup and login pages.
Included responsive design elements to ensure a seamless user experience across various devices.
Incorporated a clean and modern design for the authentication screens, following best practices.

**Functionality Implementation:**

Implemented functionality for user signup and login using email and password.
Added support for signing in with Google using Firebase Authentication.
Integrated error handling and user feedback using react-toastify for notifications.

**Firestore Integration:**

Configured Firestore to store user data during the signup process.
Ensured proper data structure and security rules for Firestore to protect user data.

**Code Formatting and Quality:**

Formatted the entire codebase for consistency and readability.
Enabled and configured ESLint to enforce coding standards and best practices.
Fixed any linting errors and warnings to maintain code quality.

**Details
Signup and Login UI:**

Created LoginPage and SignUpPage components with form validation.
Used CSS for styling and ensuring responsiveness.
Included a SignInWithGoogle button for OAuth integration.
Authentication:

Utilized firebase.auth() for email/password authentication.
Implemented signInWithPopup for Google Sign-In.
Managed authentication state using Firebase's onAuthStateChanged.
Firestore:

Integrated Firestore to save additional user data during signup.
Ensured proper user document creation and updates in Firestore.

**Code Quality:**

Added ESLint configuration and fixed existing linting issues.
Ensured code follows best practices and is consistent throughout the project.

**Testing and Verification**

Verified the signup and login flows manually to ensure proper functionality.
Tested Google Sign-In to confirm successful authentication and user data retrieval.
Checked Firestore to ensure data is correctly stored and retrieved.
Ensured responsive design works across different devices and screen sizes.

**Screenshots**

Screenshot (336)

Screenshot (337)
Screenshot (338)

**Additional Notes**
Future enhancements could include adding password reset functionality and improving error messages.
Consider adding unit and integration tests to further ensure reliability and stability.

### CODE REVIEW

File: src/Components/Sidebar/Sidebar.jsx [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+import { Link, useNavigate } from 'react-router-dom';
 import './Sidebar.css';
 import { ICONS, ACTIVE_ICONS, ACTIVE_CLASS } from '../../constants';
+import { signOut } from 'firebase/auth';
+import { auth } from '../../config/firebase.config';
 const Sidebar = () => {
+
+  const navigate = useNavigate();
+  const handleLogout = async () => {
+    await signOut(auth);
```
Comment (Jha2508) [inline src/Components/Sidebar/Sidebar.jsx]: why are we not handling error case here?

File: src/Components/SignInWithGoogle.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  async function googleLogin() {
+    const provider = new GoogleAuthProvider();
+    try {
+      const result = await signInWithPopup(auth, provider);
+      await setDoc(doc(db,"Users",result.user.uid),{
+        email: result.user.email,
+        username: result.user.displayName,
+      });
+      toast.success("Signed in with Google successfully!", {
```
Comment (Jha2508) [inline src/Components/SignInWithGoogle.jsx:21]: move strings to constant file

File: src/Components/Sidebar/Sidebar.jsx (hunk 1 above)
Comment (ShubhamTiwari55) [reply src/Components/Sidebar/Sidebar.jsx]: Done

File: src/Components/SignInWithGoogle.jsx (hunk 2 above)
Comment (ShubhamTiwari55) [reply src/Components/SignInWithGoogle.jsx:21]: Done

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**